.github/workflows/terraform-observe_scheduler.yaml

```diff
@@ -49,8 +49,38 @@ jobs:
       - uses: hashicorp/setup-terraform@v2
         with:
           terraform_version: ${{ inputs.terraform-version }}
+
+      - name: Checkout Main
+        uses: actions/checkout@v3
+        if: startsWith(${{ matrix.directory }}, 'upgrade')
+        with:
+          ref: "main"
```

Comment:
Avoid hardcoding ref names in reusable workflows. This should check out the default branch, whatever its name.